form_submit/views.py

```diff
@@ -27,40 +29,60 @@ def submit_form(request):
             language = form.cleaned_data["language"]
             file_type = form.cleaned_data["file_type"]
 
-            if os.path.basename(os.getcwd()) != "docs":
-                os.chdir("form_submit/templates/docs")
-            # TODO: add 'make clean' equivalent before 'build_sphinx'
-
-            f = open("settings.py", "w")
-            f.write("language = '" + language + "'\n")
-            f.write("link = '" + link + "'\n")
-            f.close()
-
-            if file_type == "website":
-                subprocess.call(["python", "setup.py", "build_html"])
-                return HttpResponseRedirect("/index")
-
-            elif file_type == "pdf":
-                subprocess.call(["python", "setup.py", "build_latex"])
-                os.chdir("./build/sphinx/latex")
-                pdfl = PDFLaTeX.from_texfile("AideDesignSpecs.tex")
-                pdf, log, _ = pdfl.create_pdf(keep_pdf_file=True)
+            if file_type == "validation_pdf":
+                validator = Validator()
+                validator.validate(link)
+                file_name = ".".join(
+                    validator.report_writer.report_name.split(".")[:-1] + ["pdf"]
+                )
 
-                output_file = open("AideDesignSpecs.pdf", "rb")
+                output_file = open(file_name, "rb")
 
                 response = HttpResponse(output_file, content_type="application/pdf")
 
                 response["Content-Disposition"] = (
-                    "attachment; filename=" '"AideDesignSpecs.pdf"'
+                    'attachment; filename="' + file_name + '"'
                 )
 
                 output_file.close()
 
-                # reset directory so if called again it's back in docs
-                os.chdir("../../..")
-
                 return response
 
+            else:
+                if os.path.basename(os.getcwd()) != "docs":
+                    os.chdir("form_submit/templates/docs")
+                # TODO: add 'make clean' equivalent before 'build_sphinx'
+
+                f = open("settings.py", "w")
```

Comment:
Could be more readable with a `with` statement, but it's no biggie.

Reply:
Updated!